frame/elections-phragmen/src/lib.rs

```diff
@@ -901,7 +933,9 @@ impl<T: Config> Pallet<T> {
 
 		let mut candidates_and_deposit = Self::candidates();
 		// add all the previous members and runners-up as candidates as well.
-		candidates_and_deposit.append(&mut Self::implicit_candidates_with_deposit());
+		candidates_and_deposit
+			.try_append(&mut Self::implicit_candidates_with_deposit())
+			.expect("Cannot accept more than MaxCandidates candidates.");
```

Comment:
We don't want to panic here, as `fn phragmen` runs on `on_initialize`. In practice, panicking on `on_initialize` halts the chain.

How about something along the lines of:

```suggestion
		match candidates_and_deposit
			.try_append(&mut Self::implicit_candidates_with_deposit())
			.map_err(|_| T::DbWeight::get().reads(3)) {
                Ok(_) => (),
                Err(weight) => return weight,
            }


```